example/android/app/src/androidTest/java/com/example/InstabugSample/BugReportingUITest.java

```diff
@@ -0,0 +1,176 @@
+        reproStepsDisclaimer.click();
+
+        String screenshotsListId = "instabug_vus_list";
+        assertViewWillBeVisible(screenshotsListId, 5000);
+        onView(ViewMatchers.withResourceName(screenshotsListId))
+                .check(matches(ViewMatchers.hasChildCount(2)));
+    }
+
+    @Test
+    public void onDismissCallbackIsCalled() throws InterruptedException {
+        onFlutterWidget(FlutterMatchers.withText("Set On Dismiss Callback"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+
+        onFlutterWidget(FlutterMatchers.withText("Invoke"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+        Thread.sleep(1000);
+        device.pressBack();
+
+        Thread.sleep(1000);
+
+        onFlutterWidget(FlutterMatchers.withText("onDismiss callback called with DismissType.cancel and ReportType.other"))
+                .check(FlutterAssertions.matches(FlutterMatchers.isExisting()));
+    }
+
+    @Test
+    public void changeReportTypes() throws UiObjectNotFoundException {
+        onFlutterWidget(FlutterMatchers.withText("Bug"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+        onFlutterWidget(FlutterMatchers.withText("Invoke"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+
+        // Shows bug reporting screen
+        assertViewWillBeVisible("instabug_edit_text_message", 2000);
+
+        // Close bug reporting screen
+        device.pressBack();
+        device.findObject(new UiSelector().text("DISCARD")).click();
+
+        // Enable feedback reports
+        onFlutterWidget(FlutterMatchers.withText("Feedback"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+        onFlutterWidget(FlutterMatchers.withText("Invoke"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+
+        // Shows both bug reporting and feature requests in prompt options
+        assertOptionsPromptIsVisible();
+        onView(ViewMatchers.withText("Report a bug"))
+                .check(matches(ViewMatchers.isDisplayed()));
+        onView(ViewMatchers.withText("Suggest an improvement"))
+                .check(matches(ViewMatchers.isDisplayed()));
+        onView(ViewMatchers.withText("Ask a question"))
+                .check(doesNotExist());
+    }
+
+    @Test
+    public void changeFloatingButtonEdge() {
+        onFlutterWidget(FlutterMatchers.withText("Floating Button"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+        onFlutterWidget(FlutterMatchers.withText("Move Floating Button to Left"))
+                .perform(FlutterActions.scrollTo(), FlutterActions.click());
+        onView(ViewMatchers.withResourceName("instabug_floating_button"))
+                .check(matches(isToTheLeft()));
+    }
+
+    private void assertOptionsPromptIsVisible() {
```

Comment:
**suggestion:** Moving this method before all `@Test` methods, will increase readability; considering that:
1. This class will have more tests in the future.
2. It will be close to the methods that it's used in (same as defining a variable in the nearest usage scope rule).
3. I had to search all over the place to know where this is _imported_ from.

We can also consider using a variable for `instabug_main_prompt_container` instead, what do you think?